controllers/cloudstackfailuredomain_controller_test.go

```diff
@@ -49,14 +48,83 @@ var _ = Describe("CloudStackFailureDomainReconciler", func() {
 					arg1.(*infrav1.CloudStackZoneSpec).Network.Type = cloud.NetworkTypeShared
 				}).MinTimes(1)
 
-			tempfd := &infrav1.CloudStackFailureDomain{}
-			Eventually(func() bool {
-				key := client.ObjectKeyFromObject(dummies.CSFailureDomain1)
-				if err := k8sClient.Get(ctx, key, tempfd); err == nil {
-					return tempfd.Status.Ready
-				}
-				return false
-			}, timeout).WithPolling(pollInterval).Should(BeTrue())
 		})
+
+		It("Should set failure domain Status.Ready to true.", func() {
+			assertFailureDomainCreated()
```

Comment:
`assertFailureDomainCreated` appears in the beginning of every one of these tests. Maybe we should just move it to the before function?